apps/root/src/pages/token-profile/historical-prices/index.tsx

```diff
@@ -0,0 +1,356 @@
+        ))
+      )}
+    </StyledTooltipContainer>
+  );
+};
+
+const PeriodMap: Record<Exclude<GraphContainerPeriods, GraphContainerPeriods.all>, TimeString> = {
+  [GraphContainerPeriods.day]: '30m',
+  [GraphContainerPeriods.week]: '3h',
+  [GraphContainerPeriods.month]: '12h',
+  [GraphContainerPeriods.year]: '1w',
+};
+
+const DATA_POINTS = 200;
+
+const TokenHistoricalPrices = ({ token }: TokenHistoricalPricesProps) => {
+  const mode = useThemeMode();
+  const priceService = usePriceService();
+  const providerService = useProviderService();
+  const [selectedPeriod, setSelectedPeriod] = React.useState<GraphContainerPeriods>(GraphContainerPeriods.month);
+  const prevPeriod = usePrevious(selectedPeriod);
+  const [isLoadingHistorical, setIsLoadingHistorical] = React.useState(true);
+  const [graphPrices, setGraphPrices] = React.useState<Partial<Record<GraphContainerPeriods, PriceResult[]>>>({});
+
+  const handlePeriodUpdate = (newPeriod: GraphContainerPeriods) => {
+    if (!graphPrices[newPeriod]) {
+      // When period changes, the useMemo renders no data before the isLoading is set to true by the useEffect
+      // By setting it here, we avoid the flickering of the graph
+      setIsLoadingHistorical(true);
+    }
+
+    setSelectedPeriod(newPeriod);
+  };
+
+  React.useEffect(() => {
+    const handleAllTimeframe = async () => {
+      const firstTimestamp = await providerService.getBlockTimestamp(token.chainId, 1n);
+      const currentTimestamp = Math.floor(Date.now() / 1000);
+
+      const totalDataPoints = Math.floor((currentTimestamp - firstTimestamp) / (24 * 60 * 60));
+
+      if (totalDataPoints < 30) {
+        // Fresh chain
+        return {
+          period: PeriodMap[GraphContainerPeriods.day],
+          span: DATA_POINTS,
+        };
+      }
+
+      let span = totalDataPoints;
+
+      let periodDays = 1;
+      while (span > DATA_POINTS) {
+        span = Math.floor(span / 3);
+        periodDays = periodDays * 3;
+      }
+
+      return {
+        period: `${periodDays}d` as TimeString,
+        span,
+      };
+    };
+
+    const fetchGraphPrices = async () => {
+      const priceResult = await priceService.getPricesForTokenGraph({
+        token,
+        ...(selectedPeriod === GraphContainerPeriods.all
+          ? await handleAllTimeframe()
+          : { period: PeriodMap[selectedPeriod], span: DATA_POINTS }),
+      });
+
+      setGraphPrices((prev) => ({ ...prev, [selectedPeriod]: priceResult }));
+      setIsLoadingHistorical(false);
+    };
+    if (prevPeriod !== selectedPeriod && !graphPrices[selectedPeriod]) {
+      void fetchGraphPrices();
+    }
+  }, [selectedPeriod]);
+
+  const mappedData: DataItem[] = React.useMemo(() => {
+    const prices = graphPrices[selectedPeriod];
+
+    if (!prices) return [];
+
+    const includeYear = [GraphContainerPeriods.year, GraphContainerPeriods.all].includes(selectedPeriod);
+    const format = includeYear ? 'MMM d yyyy' : 'MMM d t';
```

Comment:
based on the tickFormatter you are never going to grab the year right?

Reply:
Yes, but the design specifies x axis tick should be hidden (I agree, don't like it when its rendered)
<img width="606" alt="Screenshot 2024-08-07 at 12 23 10 PM" src="https://github.com/user-attachments/assets/b6f6d86d-c6af-4ecb-80a7-40e4597f1bef">
Date data is only displayed in the tooltip

Reply:
Mmmm I think we should display it, you can fiddle around with the tickMargin I think or some other prop, to show less of thems, but it seems important to have the dates applied

Reply:
Will do!